File: domain/src/main/java/com/ortin/gesturetranslator/domain/usecases/SaveLoadSettingsUseCase.java
package com.ortin.gesturetranslator.domain.usecases;

import com.ortin.gesturetranslator.domain.models.SettingsDomain;
import com.ortin.gesturetranslator.domain.repository.SettingsRepository;

public class SaveLoadSettingsUseCase {

    private final SettingsRepository settingsRepository;

    public SaveLoadSettingsUseCase(SettingsRepository settingsRepository) {
        this.settingsRepository = settingsRepository;
    }

    public boolean setTheme(boolean theme) {
        SettingsDomain settingsDomain = new SettingsDomain();
        settingsDomain.setTheme(theme);
        return settingsRepository.saveTheme(settingsDomain);
    }

    public boolean setGpu(boolean gpu) {
        SettingsDomain settingsDomain = new SettingsDomain();
        settingsDomain.setGpu(gpu);
        return settingsRepository.saveGpu(settingsDomain);
    }

    public boolean setPercent(boolean percent) {
        SettingsDomain settingsDomain = new SettingsDomain();
        settingsDomain.setPercent(percent);
        return settingsRepository.savePercent(settingsDomain);
    }

    public boolean setSpeedFrameDetection(int speedFrameDetection) {
        SettingsDomain settingsDomain = new SettingsDomain();
        settingsDomain.setSpeedFrameDetection(speedFrameDetection);
        return settingsRepository.saveSpeedFrameDetection(settingsDomain);
    }

    public boolean getTheme() {
        SettingsDomain settingsDomain = settingsRepository.getTheme();
        return settingsDomain.isTheme();
    }

    public boolean getGpu() {
        SettingsDomain settingsDomain = settingsRepository.getGpu();
        return settingsDomain.isGpu();
    }

    public boolean getPercent() {
        SettingsDomain settingsDomain = settingsRepository.getPercent();
        return settingsDomain.isPercent();
    }

    public int getSpeedFrameDetection() {
        SettingsDomain settingsDomain = settingsRepository.getSpeedFrameDetection();
        return settingsDomain.getSpeedFrameDetection();
    }

}

File: domain/src/main/kotlin/com/ortin/gesturetranslator/domain/usecases/SaveLoadSettingsUseCase.kt
package com.ortin.gesturetranslator.domain.usecases

import com.ortin.gesturetranslator.domain.models.SettingsDomain
import com.ortin.gesturetranslator.domain.repository.SettingsRepository

class SaveLoadSettingsUseCase(private val settingsRepository: SettingsRepository) {
    fun setTheme(theme: Boolean): Boolean {
        val settingsDomain = SettingsDomain()
        settingsDomain.setTheme(theme)
        return settingsRepository.saveTheme(settingsDomain)
    }

    fun setGpu(gpu: Boolean): Boolean {
        val settingsDomain = SettingsDomain()
        settingsDomain.setGpu(gpu)
        return settingsRepository.saveGpu(settingsDomain)
    }

    fun setPercent(percent: Boolean): Boolean {
        val settingsDomain = SettingsDomain()
        settingsDomain.setPercent(percent)
        return settingsRepository.savePercent(settingsDomain)
    }

    fun setSpeedFrameDetection(speedFrameDetection: Int): Boolean {
        val settingsDomain = SettingsDomain()
        settingsDomain.setSpeedFrameDetection(speedFrameDetection)
        return settingsRepository.saveSpeedFrameDetection(settingsDomain)
    }

    fun getTheme(): Boolean {
        val settingsDomain = settingsRepository.getTheme()
        return settingsDomain.isTheme()
    }

    fun getGpu(): Boolean {
        val settingsDomain = settingsRepository.getGpu()
        return settingsDomain.isGpu()
    }

    fun getPercent(): Boolean {
        val settingsDomain = settingsRepository.getPercent()
        return settingsDomain.isPercent()
    }

    fun getSpeedFrameDetection(): Int {
        val settingsDomain = settingsRepository.getSpeedFrameDetection()
        return settingsDomain.getSpeedFrameDetection()
    }
}